core/redis_client.py

```
@@ -0,0 +1,443 @@
+            cache_key = f"setting:{setting_key}"
+            await self.setex(cache_key, ttl, value)
+            
+            logger.debug(f"Cached setting: {setting_key}={value} (TTL={ttl}s)")
+            return True
+        except Exception as e:
+            logger.error(f"Failed to cache setting {setting_key}: {e}", exc_info=True)
+            return False
+
+    async def invalidate_setting_cache(
+        self,
+        setting_key: str = None
```

Ruff (0.14.5)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)